src/components/structures/RoomView.js

```diff
@@ -504,17 +526,21 @@ module.exports = React.createClass({
         }
     },
 
-    _paginateCompleted: function() {
-        debuglog("paginate complete");
-
-        // we might have switched rooms since the paginate started - just bin
+    _onTimelineUpdated: function(gotResults) {
+        // we might have switched rooms since the load started - just bin
         // the results if so.
         if (this.unmounted) return;
 
         this.setState({
-            room: MatrixClientPeg.get().getRoom(this.props.roomId),
             paginating: false,
         });
+
+        if (gotResults) {
+            this.setState({
+                events: this._timelineWindow.getEvents(),
```

Comment:
I don't like having multiple sources of truth. Fine you're just holding onto a reference, but why not just directly call `this._timelineWindow.getEvents()` when you need the events? I'm worried that this current impl will add more book-keeping.